packages/core/src/new-api/internal/deployer.ts

```diff
@@ -26,30 +27,45 @@ import {
   DeploymentExecutionState,
   ExecutionStateMap,
 } from "./types/execution-state";
+import { TransactionLookupTimer } from "./types/transaction-timer";
 import { assertIgnitionInvariant } from "./utils/assertions";
 import { getFuturesFromModule } from "./utils/get-futures-from-module";
 import { validate } from "./validation/validate";
 
-const BLOCK_POLLING_INTERVAL = 200;
+// The interval between checks to see if a new block has been created
+const DEFAULT_BLOCK_POLLING_INTERVAL = 200;
+
+// In milliseconds the amount of time to wait on a transaction to
+// confirm before timing out
+const DEFAULT_TRANSACTION_TIMEOUT_INTERVAL = 3 * 60 * 1000;
 
 /**
  * Run an Igntition deployment.
  *
  * @beta
  */
 export class Deployer {
+  private _config: DeployConfig;
   private _moduleConstructor: ModuleConstructor;
   private _executionEngine: ExecutionEngine;
   private _strategy: ExecutionStrategy;
   private _artifactResolver: ArtifactResolver;
   private _deploymentLoader: DeploymentLoader;
   private _chainDispatcher: ChainDispatcher;
+  private _transactionLookupTimer: TransactionLookupTimer;
```

Comment:
This _class_ (God forgive me) encapsulates setting times against transaction lookup starts so that timeouts can be determined.